mavros/src/plugins/imu_pub.cpp

```diff
@@ -203,9 +203,9 @@ class IMUPubPlugin : public MavRosPlugin {
 		mavlink_attitude_t att;
 		mavlink_msg_attitude_decode(msg, &att);
 
-		auto orientation = UAS::transform_frame_ned_enu(
+		auto orientation = UAS::transform_frame_platform_enu(
 				UAS::quaternion_from_rpy(att.roll, att.pitch, att.yaw));
-		auto gyro = UAS::transform_frame_ned_enu(
+		auto gyro = UAS::transform_frame_platform_enu(
```

Comment:
You're absolutely correct actually.

I came to the conclusion that a static rotation matrix isn't quite right.
Your approach seems correct to me.
On 20 Jan 2016 7:59 pm, "Eddy Scott" <notifications@github.com> wrote:

> In mavros/src/plugins/imu_pub.cpp
> <https://github.com/mavlink/mavros/pull/473#discussion_r50259647>:
>
> >  				UAS::quaternion_from_rpy(att.roll, att.pitch, att.yaw));
> > -		auto gyro = UAS::transform_frame_ned_enu(
> > +		auto gyro = UAS::transform_frame_platform_enu(
>
> This still confuses me. The only time that transforming data expressed in
> the body-fixed frame into the ENU frame with a roll of 180 degrees is
> correct is in when the body-fixed frame's X axis is aligned with East.
>
> In fact, data that is *truly* expressed in the body-fixed frame should
> never be rotated by a static quaternion, it should be rotated by the
> attitude calculated by the FCU. Since the attitude represents the dynamic
> transform from the body-fixed frame to a fixed inertial frame (e.g. NED or
> ENU), the attitude should be rotated by a static transform to facilitate
> different frame expressions.
>
> For example consider the case where the acceleration data is given in the
> body-fixed frame, and the FCU computes the attitude of the body-fixed frame
> with respect to the NED frame. If the user wants the acceleration data
> reported in the ENU frame they need to rotate the body-fixed frame
> quaternion to now express the attitude of the vehicle with respect to the
> ENU frame. They then need to transform the acceleration data with this new
> quaternion.
>
> I think that this should be the fundamental way transformations from the
> body-fixed frame should work. If data is already expressed in the NED frame
> (e.g. the velocities reported by the local_position plugin) then a static
> transform should be applied.
>
> Am I wrong here?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mavlink/mavros/pull/473/files#r50259647>.
>